File: app_manager/appmanager/src/main/java/app/editors/manager/ui/fragments/main/DocsCommonFragment.java
package app.editors.manager.ui.fragments.main;

import android.os.Bundle;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import app.editors.manager.managers.providers.CloudFileProvider;

public class DocsCommonFragment extends DocsCloudFragment {

    public static DocsCommonFragment newInstance() {
        return new DocsCommonFragment();
    }

    public static final String ID = CloudFileProvider.Section.Common.getPath();

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        init();
    }

    @Override
    protected boolean onSwipeRefresh() {
        if (!super.onSwipeRefresh()) {
            mCloudPresenter.getItemsById(ID);
            return true;
        }

        return false;
    }

    @Override
    public void onScrollPage() {
        super.onScrollPage();
        if  (mCloudPresenter.getStack() == null){
            mCloudPresenter.getItemsById(ID);
        }
    }

    @Override
    public void onStateEmptyBackStack() {
        super.onStateEmptyBackStack();
        if (mSwipeRefresh != null) {
            mSwipeRefresh.setRefreshing(true);
        }
        mCloudPresenter.getItemsById(ID);
    }

    @Override
    public void onRemoveItemFromFavorites() {

    }

    private void init() {
        mCloudPresenter.checkBackStack();
    }

}

File: app_manager/appmanager/src/main/java/app/editors/manager/ui/fragments/main/DocsCommonFragment.kt
package app.editors.manager.ui.fragments.main

import android.os.Bundle
import android.view.View
import app.documents.core.network.ApiContract
import app.editors.manager.managers.providers.CloudFileProvider

class DocsCommonFragment : DocsCloudFragment() {

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        init()
    }

    override fun onSwipeRefresh(): Boolean {
        if (!super.onSwipeRefresh()) {
            cloudPresenter.getItemsById(ID)
            return true
        }
        return false
    }

    override fun onScrollPage() {
        super.onScrollPage()
        if (cloudPresenter.stack == null) {
            cloudPresenter.getItemsById(ID)
        }
    }

    override fun onStateEmptyBackStack() {
        super.onStateEmptyBackStack()
        swipeRefreshLayout?.isRefreshing = true
        cloudPresenter.getItemsById(ID)
    }

    override fun onRemoveItemFromFavorites() {}

    private fun init() {
        cloudPresenter.checkBackStack()
    }

    override val section
        get() = ApiContract.SectionType.CLOUD_COMMON

    companion object {
        val ID = CloudFileProvider.Section.Common.path

        fun newInstance(account: String): DocsCommonFragment {
            return DocsCommonFragment().apply {
                arguments = Bundle(1).apply {
                    putString(KEY_ACCOUNT, account)
                }
            }
        }
    }
}